src/component/parser/json/converter/DiagramConverter.ts

```diff
@@ -136,26 +153,37 @@ export default class DiagramConverter {
 
   private deserializeEdges(edges: BPMNEdge | BPMNEdge[]): Edge[] {
     return ensureIsArray(edges)
-      .map(edge => {
+      .map(bpmnEdge => {
         const flow =
-          this.convertedElements.findSequenceFlow(edge.bpmnElement) ||
-          this.convertedElements.findMessageFlow(edge.bpmnElement) ||
-          this.convertedElements.findAssociationFlow(edge.bpmnElement);
+          this.convertedElements.findSequenceFlow(bpmnEdge.bpmnElement) ||
+          this.convertedElements.findMessageFlow(bpmnEdge.bpmnElement) ||
+          this.convertedElements.findAssociationFlow(bpmnEdge.bpmnElement);
 
         if (!flow) {
-          this.parsingMessageCollector.warning(new EdgeUnknownBpmnElementWarning(edge.bpmnElement));
+          this.parsingMessageCollector.warning(new EdgeUnknownBpmnElementWarning(bpmnEdge.bpmnElement));
           return;
         }
 
-        const waypoints = this.deserializeWaypoints(edge.waypoint);
-        const label = this.deserializeLabel(edge.BPMNLabel, edge.id);
-        const messageVisibleKind = edge.messageVisibleKind ? (edge.messageVisibleKind as unknown as MessageVisibleKind) : MessageVisibleKind.NONE;
+        const waypoints = this.deserializeWaypoints(bpmnEdge.waypoint);
+        const label = this.deserializeLabel(bpmnEdge.BPMNLabel, bpmnEdge.id);
+        const messageVisibleKind = bpmnEdge.messageVisibleKind ? (bpmnEdge.messageVisibleKind as unknown as MessageVisibleKind) : MessageVisibleKind.NONE;
 
-        return new Edge(edge.id, flow, waypoints, label, messageVisibleKind);
+        const edge = new Edge(bpmnEdge.id, flow, waypoints, label, messageVisibleKind);
+        DiagramConverter.setColorExtensionsOnEdge(edge, bpmnEdge);
+        return edge;
       })
       .filter(Boolean);
   }
 
+  // 'BPMN in Color' extensions  with fallback to bpmn.io colors
+  private static setColorExtensionsOnEdge(edge: Edge, bpmnEdge: BPMNEdge): void {
+    if ('border-color' in bpmnEdge) {
+      edge.extensions['bv:color:stroke'] = <string>bpmnEdge['border-color'];
```

Comment:
**issue:** 

It's easy to make a mistake with the property name `edge.extensions['bv:color:stroke']` due to its current implementation.

Reply:
I don't think so. Try it in an IDE an you will get type hints while typing.
There are places in the code and tests where we use such patterns.